### DISCUSSION TITLE
Exclude App_Data folder from publishing

### DISCUSSION THREAD

**Discussion by DavidStania** [OrchardCMS/OrchardCore#12487] [Q&A] [answered]
Hey all,

I made a lot of try's to exclude the App_Data folder from the publishing process. I can't find any solution how to exclude them. OrchardCode used this folder for the tenant management. If I have a lot of tenants on my webserver eg. azure, with the next publishing all data and the tenants.json file will be overwriten and this destroys my page.

It would be greate, if someone have a good idea how to and the solution would be published in the documentation.

Best regards
David

**Comment by Skrypt** [comment]
Look at the .csproj publishing exclude rules. It is documented by Microsoft.

https://learn.microsoft.com/en-us/aspnet/web-forms/overview/deployment/advanced-enterprise-web-deployment/excluding-files-and-folders-from-deployment

**Reply by Skrypt** [reply]
Else you can alternatively use Azure Storage for your tenant configs. This is documented.

**Reply by DavidStania** [reply]
The Azure Storage is a good alternative, you are right. Tank you.

The exclude rules like documented by Microsoft is - I belive - not working with .Net Core Applications. I tryed them out with out success.

**Reply by DavidStania** [reply]
Done it with Azure Storage. Thank you!!

**Comment by jtkech** [comment, marked as answer]
@DavidStania 

Just for info for another possible solution in OrchardCore, look at #4402.

So in your case try to add the following property in your app project file

    <PropertyGroup>
       ...
      <DefaultItemExcludes>$(DefaultItemExcludes);App_Data\**</DefaultItemExcludes>
    </PropertyGroup>

**Reply by DavidStania** [reply]
Incredible! It is working. Seriously, I was looking since a long time for this solution.
